solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java

```diff
@@ -16,73 +16,217 @@
  */
 package org.apache.solr.ltr.model;
 
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
 import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.solr.core.SolrResourceLoader;
 import org.apache.solr.ltr.feature.FeatureException;
 import org.apache.solr.ltr.norm.IdentityNormalizer;
 import org.apache.solr.ltr.norm.Normalizer;
 import org.apache.solr.ltr.ranking.Feature;
+import org.apache.solr.ltr.ranking.Feature.FeatureWeight;
+import org.apache.solr.ltr.ranking.FeatureWeightCreator;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.util.SolrPluginUtils;
 
 /**
  * Contains all the data needed for loading a model.
  */
 
 public abstract class LTRScoringModel {
 
+  public static class ModelFeature  implements FeatureWeightCreator {
```

Comment:
Let's rip out the merging of normalizers + features from this pluggable scoring models pull request so we don't hold up the other tickets dependent on pluggable models with this added change.